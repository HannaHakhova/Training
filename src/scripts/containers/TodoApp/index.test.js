import React from 'react';
import {func} from 'prop-types';
import {shallow, spy, mount} from 'enzyme';
import TodoApp from './index';

const mockTodoList = jest.mock('components/TodoList', () => {
    const todo = {
        id: 1,
        text: 'Do task1',
        isDone: false
    };
    const TodoList = ({toggle}) => <div id={'mockId'} onClick={() => { toggle(todo); }}/>;
    TodoList.propTypes = {
        toggle: func
    };
    return TodoList;
});

describe('TodoApp', () => {
    it('TodoApp match expected snapshot', () => {
        expect(shallow(<TodoApp/>)).toMatchSnapshot();
    });
});

it('calls toggle the button click', () => {
    mockTodoList.find('button').simulate('click');
    expect(mockTodoList.state(mockTodoList.id).value).toEqual({
        id: 1,
        text: 'Do task1',
        isDone: true
    });
});

