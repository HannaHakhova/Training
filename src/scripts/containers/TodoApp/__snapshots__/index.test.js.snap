// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoPage TodoPage match expected snapshot 1`] = `
<div>
  <NewTodo
    className="container"
    createTask={[Function]}
  />
  <p>
    I  need to do 0 tasks.
  </p>
  <TodoList
    todos={
      Array [
        Object {
          "id": 1,
          "isDone": false,
          "text": "Do task1",
        },
        Object {
          "id": 2,
          "isDone": false,
          "text": "Do task2",
        },
        Object {
          "id": 3,
          "isDone": false,
          "text": "Do task3",
        },
      ]
    }
    toggle={[Function]}
  />
</div>
`;
